lib/src/Pos/Logic/Pure.hs

```diff
@@ -0,0 +1,314 @@
+block :: Block
+block = Right mainBlock
+
+mainBlock :: MainBlock
+mainBlock = UnsafeGenericBlock
+    { _gbHeader = mainBlockHeader
+    , _gbBody   = blockBody
+    , _gbExtra  = extraBodyData
+    }
+
+blockBody :: Body MainBlockchain
+blockBody = MainBody
+    { _mbTxPayload     = emptyTxPayload
+    , _mbSscPayload    = emptySscPayload
+    , _mbDlgPayload    = emptyDlgPayload
+    , _mbUpdatePayload = emptyUpdatePayload
+    }
+
+emptyTxPayload :: TxPayload
+emptyTxPayload = UnsafeTxPayload
+    { _txpTxs       = MerkleEmpty
+    , _txpWitnesses = []
+    }
+
+-- SscPayload is 4 alternatives. I chose CertificatesPayload because it has
+-- the fewest fields...
+emptySscPayload :: SscPayload
+emptySscPayload = CertificatesPayload
+    { spVss = UnsafeVssCertificatesMap
+          { getVssCertificatesMap = mempty
+          }
+    }
+
+emptyDlgPayload :: DlgPayload
+emptyDlgPayload = UnsafeDlgPayload
+    { getDlgPayload = []
+    }
+
+emptyUpdatePayload :: UpdatePayload
+emptyUpdatePayload = UpdatePayload
+    { upProposal = Nothing
+    , upVotes    = []
+    }
+
+extraBodyData :: ExtraBodyData MainBlockchain
+extraBodyData = MainExtraBodyData
+    { _mebAttributes = Attributes
+          { attrData   = ()
+          , attrRemain = UnparsedFields mempty
+          }
+    }
+
+blockHeader :: BlockHeader
+blockHeader = Right mainBlockHeader
+
+mainBlockHeader :: MainBlockHeader
+mainBlockHeader = UnsafeGenericBlockHeader
+    { _gbhPrevBlock = mainBlockHeaderHash
+    , _gbhBodyProof = bodyProof
+    , _gbhConsensus = consensusData
+    , _gbhExtra     = extraHeaderData
+    }
+
+mainBlockHeaderHash :: HeaderHash
+mainBlockHeaderHash = AbstractHash (Crypto.hash (mempty :: ByteString))
+
+bodyProof :: BodyProof MainBlockchain
+bodyProof = MainProof
+    { mpTxProof       = txProof
+    , mpMpcProof      = sscProof
+    , mpProxySKsProof = dlgProof
+    , mpUpdateProof   = updateProof
+    }
+
+txProof :: TxProof
+txProof = TxProof
+    { txpNumber        = 0
+    , txpRoot          = MerkleRoot (AbstractHash (Crypto.hash (mempty :: ByteString)))
+    , txpWitnessesHash = AbstractHash (Crypto.hash (mempty :: ByteString))
+    }
+
+sscProof :: SscProof
+sscProof = CertificatesProof
+    { sprVss = AbstractHash (Crypto.hash (mempty :: ByteString))
+    }
+
+dlgProof :: Hash DlgPayload
+dlgProof = AbstractHash (Crypto.hash (mempty :: ByteString))
+
+updateProof :: UpdateProof
+updateProof = AbstractHash (Crypto.hash (mempty :: ByteString))
+
+consensusData :: ConsensusData MainBlockchain
+consensusData = MainConsensusData
+    { _mcdSlot       = slotId
+    , _mcdLeaderKey  = publicKey
+    , _mcdDifficulty = chainDifficulty
+    , _mcdSignature  = blockSignature
+    }
+
+slotId :: SlotId
+slotId = SlotId
+    { siEpoch = EpochIndex { getEpochIndex = 0 }
+    , siSlot  = UnsafeLocalSlotIndex { getSlotIndex = 0 }
+    }
+
+publicKey :: PublicKey
+publicKey = PublicKey key
+  where
+    Right key = xpub (BS.replicate 64 0)
```

Comment:
I don't like usage of so low-level functions. Why not use `fst . deterministicKeyGen` to create a `PublicKey`?

Reply:
I checked `deterministicKeyGen`. It uses `createKeypairFromSeed` which has a comment:

> TODO: this is just a placeholder for actual (not ready yet) derivation
> of keypair from seed in cardano-crypto API

Should I really use that?

Also, it's a bit different. Here I know that I have 64 zeros as the xpub. If I use `deterministicKeyGen` I don't know up-front what's in there. Not that it matters really, it's just, why introduce (pseudo)randomness when it's not needed?

Reply:
> Should I really use that?

I don't know what that comment means, but `keyGen` is used in production, so I think it's better to use this function wherever possible (if there are any problems with it, using this function in more places increases chances of discovering them).

> Here I know that I have 64 zeros as the xpub. If I use deterministicKeyGen I don't know up-front what's in there.

Why do you need to know it? The less you know about xpub, the better. You shouldn't ever use `XPub` directly, except some modules in `cardano-sl-crypto`. Here you rely on the size of public key.

> Not that it matters really, it's just, why introduce (pseudo)randomness when it's not needed?

The reason is the same as for hashes.

Reply:
> Why do you need to know it? The less you know about xpub, the better.

I don't need to know it, but I also don't need it to be random.

> Here you rely on the size of public key.

Fair point, I'll make the change.